src/ripple/core/tests/Config.test.cpp

```diff
@@ -0,0 +1,356 @@
+3
+
+# Turn down default logging to save disk space in the long run.
+# Valid values here are trace, debug, info, warning, error, and fatal
+[rpc_startup]
+{ "command": "log_level", "severity": "warning" }
+
+# Defaults to 1 ("yes") so that certificates will be validated. To allow the use
+# of self-signed certificates for development or internal use, set to 0 ("no").
+[ssl_verify]
+0
+
+[sqdb]
+backend=sqlite
+)rippleConfig");
+
+    if (!dbPath.empty ())
+        return boost::str (configContentsTemplate % "[database_path]" % dbPath);
+    else
+        return boost::str (configContentsTemplate % "" % "");
+}
+
+/**
+   Write a config file and remove when done.
+ */
+class ConfigGuard
+{
+private:
+    using path = boost::filesystem::path;
+    path subDir_;
+    path configFile_;
+    path dataDir_;
+
+    bool rmSubDir_{false};
+    bool rmDataDir_{false};
+
+    Config config_;
+
+public:
+    ConfigGuard (std::string subDir, std::string const& dbPath)
+        : subDir_ (std::move (subDir)), dataDir_ (dbPath)
+    {
+        using namespace boost::filesystem;
+
+        if (dbPath.empty ())
+        {
+            dataDir_ = subDir_ / path (Config::Helpers::getDatabaseDirName ());
+        }
+
+        configFile_ = subDir_ / path (Config::Helpers::getConfigFileName ());
+        {
+            if (!exists (subDir_))
+            {
+                create_directory (subDir_);
+                rmSubDir_ = true;
+            }
+            else if (is_directory (subDir_))
+                rmSubDir_ = false;
+            else
+            {
+                // Cannot run the test someone created a file where we want to
+                // put out directory
+                throw std::runtime_error ("Cannot create directory: " +
+                                          subDir_.string ());
+            }
+        }
+
+        if (!exists (configFile_))
+        {
+            std::ofstream o (configFile_.string ());
+            o << configContents (dbPath);
+        }
+        else
+        {
+            throw std::runtime_error (
+                "Refusing to overwrite existing config file: " +
+                configFile_.string ());
+        }
+
+        rmDataDir_ = !exists (dataDir_);
+        config_.setup (configFile_.string (), /*bQuiet*/ false);
+    }
+    Config& config ()
+    {
+        return config_;
+    }
+    bool dataDirExists () const
+    {
+        return boost::filesystem::is_directory (dataDir_);
+    }
+    bool configFileExists () const
+    {
+        return boost::filesystem::is_regular_file (configFile_);
+    }
+    ~ConfigGuard ()
+    {
+        try
+        {
+            using namespace boost::filesystem;
+            if (!is_regular_file (configFile_))
+                std::cerr << "Expected " << configFile_.string ()
+                          << " to be an existing file.\n";
+            else
+                remove (configFile_.string ());
+
+            auto rmDir = [](path const& toRm)
```

Comment:
We're inclined to allow this particular usage of the lambda since its buried away in a unit test. However, in business logic code we would want this to be a member or free function. Rationale: over time, these lambdas can only increase in size. Eventually one will crash and we'll be left with an anonymously named lambda whose mangled name does not encode its argument types. The lambda has no name that we can perform full-text source searches, and IDEs typically provide less information about them than they do named functions.